### ISSUE TITLE
Gallery Block + Link to = Crop images doesn't work in the editor

### ISSUE THREAD

**Issue by gyurmey2** [WordPress/gutenberg#56286]
### Description

Crop images doesn't work in the editor if you choose linking to Attachemnt Page or Media File.

### Step-by-step reproduction instructions

1. Add a gallery block
2. Add images (preferably of different proportions)
3. Select **Crop images**
4. Select **Link to** Attachemnt Page or Media File

### Screenshots, screen recording, code snippet

![at](https://github.com/WordPress/gutenberg/assets/113413673/024ff4ce-32ef-4849-ba16-5936d5f4f6aa)

### Environment info

- WP: 6.4.1
- Theme: TT4

### Please confirm that you have searched existing issues in the repo.

Yes

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

Yes

**Comment by michalczaplinski** [comment]
I can't seem to replicate this with TT4 & 6.4.1. Which version of the Gutenberg plugin are you running?


https://github.com/WordPress/gutenberg/assets/5417266/72342e1a-4c28-4746-a980-08c20bb86e82

**Comment by t-hamano** [comment]
I was able to reproduce this problem with WordPress6.4, but not with the latest version of Gutenberg.

https://github.com/WordPress/gutenberg/assets/54422211/cddff5fa-838f-4fe2-b2cf-775cf5cd6120

My guess is that #56043 may have unintentionally solved this problem. Due to #56043, the HTML of the Image block in the Gallery block has changed as follows.

WordPress 6.4

```html
<figure class="wp-block-image">
	<a>
		<div>
			<img />
		</div>
	</a>
	<figcaption></figcaption>
</figure>
```

Latest Gutenberg trunk

```html
<figure class="wp-block-image">
	<div>
		<a>
			<img />
		</a>
	</div>
	<figcaption></figcaption>
</figure>
```

@gyurmey2 I would like to close this issue as I believe it has been resolved.

This fix will be part of Gutenberg plugin 17.2. It hasn't been released yet, but please give it a try once it's released.
